### DISCUSSION TITLE
NanoPi R2S how to enable UART1 ?

### DISCUSSION THREAD

**Discussion by Mysteo91** [Ysurac/openmptcprouter#1879] [Q&A]
Hello!  I'm using NanoPi R2S and openmptcprouter. I found that is not possible to send data via ttyS1, I think UART1 is disabled in the kernel. How do I create an openmptcprouter with UART1 enabled ?

**Comment by Ysurac** [comment]
No idea. You can check on OpenWrt if you find something.

**Reply by Mysteo91** [reply]
I solved my problem by rebuilding the kernel.
In the target / linux / rockchip / patches-5.4 / 001-rockchip-rk3328-Add-support-for-FriendlyARM-NanoPi-R.patch file, I added a line:

```
+ & uart1 {
+ status = "okay";
+};
```
Now UART1 is working
